File: PROGRAMMERS/LEVEL2/더 맵게.cpp
#include<iostream>
#include<string>
#include<vector>
#include<algorithm>
#include<queue>

using namespace std;
int solution(vector<int> scoville, int K) {
    int answer = 0;
    priority_queue<int, vector<int>, greater<int>> q;
    for (int i = 0; i < scoville.size(); i++) {
        q.push(scoville[i]);
    }
    while (q.top() < K && q.size() > 1) {
        int first = q.top();
        q.pop();
        int second = q.top();
        q.pop();
        q.push(first + 2 * second);
        answer++;

    }
    if (q.top() < K)return -1;


    return answer;
}

File: PROGRAMMERS/LEVEL2/더 맵게.java
import java.util.Collections;
import java.util.PriorityQueue;
import java.util.Queue;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        Queue<Integer> pq = new PriorityQueue<Integer>();
        for(int x : scoville) {
        	pq.add(x);
        }
        
        while(pq.peek()<K && pq.size()>1) {
        	int num1 = pq.poll();
        	int num2 = pq.poll();
        	
        	pq.add(num1+(num2*2));
        	answer++;
        }
        
        if(pq.peek()<K) return -1;
        
        return answer;
    }
}